File: LivePriceBackend/DTOs/CParityRule/CParityRuleCreateModel.cs
using LivePriceBackend.Constants.Enums;

namespace LivePriceBackend.DTOs.CParityRule;

public class CParityRuleCreateModel
{
    public int CustomerId { get; set; }
    public int ParityId { get; set; }
    public bool IsVisible { get; set; }
    public SpreadRuleType? SpreadRuleType { get; set; }
    public decimal? SpreadForAsk { get; set; }
    public decimal? SpreadForBid { get; set; }
}

File: live_price_frontend/lib/modules/customer_panel/models/c_parity_rule_create_model.dart
import 'c_parity_rule_view_model.dart';

class CParityRuleCreateModel {
  final int customerId;
  final int parityId;
  final bool isVisible;
  final SpreadRuleType? spreadRuleType;
  final double? spreadForAsk;
  final double? spreadForBid;

  CParityRuleCreateModel({
    required this.customerId,
    required this.parityId,
    required this.isVisible,
    this.spreadRuleType,
    this.spreadForAsk,
    this.spreadForBid,
  });

  Map<String, dynamic> toJson() {
    return {
      'customerId': customerId,
      'parityId': parityId,
      'isVisible': isVisible,
      'spreadRuleType': spreadRuleType?.id,
      'spreadForAsk': spreadForAsk,
      'spreadForBid': spreadForBid,
    };
  }
}
